src/pykeen/datasets/__init__.py

```diff
@@ -189,6 +192,45 @@ def get_dataset(
     )
 
 
+def _digest_kwargs(dataset_kwargs: Mapping[str, Any]) -> str:
+    digester = hashlib.sha256()
+    for key in sorted(dataset_kwargs.keys()):
+        digester.update(key.encode(encoding="utf8"))
+        digester.update(str(dataset_kwargs[key]).encode(encoding="utf8"))
+    return base64.urlsafe_b64encode(digester.digest()).decode("utf8")[:32]
+
+
+def _cached_get_dataset(
+    dataset: str,
+    dataset_kwargs: Optional[Mapping[str, Any]],
+    force: bool = False,
+) -> Dataset:
+    """Get dataset by name, potentially using file-based cache."""
+    # hash kwargs
+    dataset_kwargs = dataset_kwargs or {}
+    digest = _digest_kwargs(dataset_kwargs)
+
+    # normalize dataset name
+    dataset = dataset_resolver.normalize(dataset)
+
+    # get canonic path
+    path = PYKEEN_DATASETS.joinpath(dataset, "cache", digest)
+
+    # try to use cached dataset
+    if path.is_dir() and not force:
+        logger.info(f"Loading cached preprocessed dataset from {path.as_uri()}")
+        return Dataset.from_directory_binary(path)
```

Comment:
@cthoyt If the dataset is loaded from the cached version, it will not be of a specific class, but rather always an `EagerDataset`.

Reply:
Is this a problem? Perhaps we also need to make sure that it knows what the name of the dataset and original class are?

Reply:
Maybe anywhere there's a `get_normalized_name` usage on a dataset? We should probably retire some of those functions...

Reply:
7ef78360d2e5adb5f594d6a22d849a6fd42654fa